# USING YOLO

Yolo is a Computer Vision library that identifies and localizates objects in an image, fortunately the hardwork was done before in the boxing task at [roboflow](roboflow.com), now all we have to do it's to train our model by ourselves.

![Image](../LOGSBIN/assets/doom_attack_57_jpg.rf.aface71250b423830e724ec10f18e0ef.jpg)
>[!NOTE]
>Notice that YOLO uses an image detection resolution of 448x448, that's why the image looks like that, the boxing is going to be found at labels file

In [10]:
import os
from dotenv import load_dotenv

load_dotenv()

api_key = os.getenv("ROBOFLOW_API_KEY")
workspace = os.getenv("WORKSPACE")

In [12]:
from roboflow import Roboflow
rf = Roboflow(api_key=api_key)
project = rf.workspace(workspace).project("doom-enemy-classifier")
dataset = project.version(1).download("yolov11")

loading Roboflow workspace...
loading Roboflow project...
Exporting format yolov11 in progress : 0.1%
Version export complete for yolov11 format



Extracting Dataset Version Zip to Doom-Enemy-Classifier-1 in yolov11:: 100%|██████████| 584/584 [00:00<00:00, 9811.16it/s]


### Here comes the real training 
A YOLO model only need a few lines and a little time to get the model trained and we are going to use a public chekpoint to make the training faster.

In [14]:
from ultralytics import YOLO

model = YOLO("yolo11n.pt")

results = model.train(
    data = f"{dataset.location}/data.yaml",
    epochs = 30,
    imgsz = 640,
    cache = True,
    device = "mps" #IMPORTANT: If you are using a Nvidia GPU just comment this kwarg
)

Ultralytics 8.4.155 🚀 Python-3.14.3 torch-2.14.0 MPS (Apple M4)
engine/trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=True, cfg=None, channels_last=None, classes=None, close_mosaic=10, cls=0.5, cls_pw=0.0, cls_remap=True, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=/Users/emmanuel/DeVeLoP/AI-ML/doom-mlg-ai/YOLOMODEL/Doom-Enemy-Classifier-1/data.yaml, degrees=0.0, deterministic=True, device=mps, dfl=1.5, dgrad=0.5, dis=6.0, distill_model=None, dlam=1.0, dlog=1.0, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=30, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo11n.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=train-2, 

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


       1/30      4.25G      1.551      4.349      1.307         15        640: 100% ━━━━━━━━━━━━ 14/14 1.2s/it 16.9s0.8s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 1.4s/it 1.4s
                   all         20         34     0.0624      0.801      0.187      0.101

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       2/30      4.25G      1.334      3.934      1.146         36        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


       2/30      5.22G      1.514      3.576      1.313         32        640: 100% ━━━━━━━━━━━━ 14/14 1.5it/s 9.1s0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.7it/s 0.4s
                   all         20         34     0.0179      0.942      0.225      0.127

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       3/30      4.32G      1.598      3.088      1.369         35        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


       3/30      5.25G      1.535      3.131      1.345         26        640: 100% ━━━━━━━━━━━━ 14/14 1.6it/s 8.7s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.6it/s 0.4s
                   all         20         34     0.0108      0.798      0.189     0.0643

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       4/30      4.25G      1.539      3.244      1.407         32        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


       4/30      5.22G      1.558      2.976      1.377         27        640: 100% ━━━━━━━━━━━━ 14/14 1.6it/s 8.5s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 3.0it/s 0.3s
                   all         20         34     0.0193      0.914      0.204     0.0756

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       5/30      4.32G      1.292      3.054      1.291         21        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


       5/30      5.22G      1.501       2.85      1.368         24        640: 100% ━━━━━━━━━━━━ 14/14 1.6it/s 8.5s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.9it/s 0.3s
                   all         20         34     0.0136      0.816      0.218     0.0778

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       6/30      4.32G      1.475      2.448      1.275         40        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


       6/30      5.26G      1.466      2.646      1.355         17        640: 100% ━━━━━━━━━━━━ 14/14 1.7it/s 8.4s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.9it/s 0.3s
                   all         20         34      0.419      0.174      0.143     0.0582

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       7/30      4.25G       1.41      2.481      1.276         46        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


       7/30      5.25G      1.448      2.424      1.349         22        640: 100% ━━━━━━━━━━━━ 14/14 1.6it/s 8.8s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.2it/s 0.5s
                   all         20         34      0.955     0.0714      0.177     0.0892

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       8/30      4.25G      1.455      2.358      1.312         40        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


       8/30      5.22G      1.399      2.386      1.307         28        640: 100% ━━━━━━━━━━━━ 14/14 1.5it/s 9.1s0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.6it/s 0.4s
                   all         20         34       0.69      0.119      0.292      0.155

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       9/30      4.33G      1.631      2.432      1.472         33        640: 0% ──────────── 0/14  0.8s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


       9/30      5.25G      1.461      2.326      1.318         35        640: 100% ━━━━━━━━━━━━ 14/14 1.6it/s 8.9s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.8it/s 0.4s
                   all         20         34       0.27      0.417      0.434      0.224

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      10/30      4.25G      1.476      2.153      1.367         41        640: 0% ──────────── 0/14  0.8s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      10/30      4.26G      1.383      2.112      1.268         26        640: 100% ━━━━━━━━━━━━ 14/14 1.6it/s 8.7s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.9it/s 0.3s
                   all         20         34      0.321      0.521      0.516      0.278

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      11/30      4.25G       1.15      1.993      1.211         33        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      11/30      5.22G      1.339      2.031       1.26         34        640: 100% ━━━━━━━━━━━━ 14/14 1.6it/s 8.6s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.9it/s 0.3s
                   all         20         34      0.487      0.375      0.511      0.245

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      12/30      4.31G      1.186      1.876      1.197         33        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      12/30      5.22G      1.402      2.091       1.32         32        640: 100% ━━━━━━━━━━━━ 14/14 1.6it/s 8.8s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.8it/s 0.4s
                   all         20         34      0.554      0.511      0.501      0.292

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      13/30      5.22G      1.327      1.978      1.273         30        640: 100% ━━━━━━━━━━━━ 14/14 1.5it/s 9.5s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.8it/s 0.4s
                   all         20         34      0.659      0.673      0.708      0.399

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      14/30      4.32G      1.101      1.843      1.206         37        640: 0% ──────────── 0/14  0.8s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      14/30      5.22G      1.313      1.979      1.276         26        640: 100% ━━━━━━━━━━━━ 14/14 1.5it/s 9.1s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.8it/s 0.4s
                   all         20         34      0.892      0.602      0.737      0.399

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      15/30      5.26G      1.295      1.789      1.245         26        640: 100% ━━━━━━━━━━━━ 14/14 1.6it/s 8.8s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.6it/s 0.4s
                   all         20         34      0.769      0.697      0.764      0.459

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      16/30      4.25G      1.179      1.889      1.167         31        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      16/30      5.25G      1.287      1.825      1.239         23        640: 100% ━━━━━━━━━━━━ 14/14 1.4it/s 9.8s0.8s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.5it/s 0.4s
                   all         20         34      0.802      0.733      0.797      0.485

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      17/30      4.26G      1.207      2.267      1.259         24        640: 0% ──────────── 0/14  0.8s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      17/30      5.22G      1.268      1.777      1.216         29        640: 100% ━━━━━━━━━━━━ 14/14 1.4it/s 9.9s0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.8it/s 0.4s
                   all         20         34      0.822       0.71      0.744      0.447

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      18/30      5.26G      1.213      1.619      1.188         40        640: 100% ━━━━━━━━━━━━ 14/14 1.4it/s 10.1s0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 1.8it/s 0.6s
                   all         20         34      0.699      0.751      0.741      0.431

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      19/30      4.26G      1.259      1.626      1.234         20        640: 100% ━━━━━━━━━━━━ 14/14 1.4it/s 10.0s0.8s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.5it/s 0.4s
                   all         20         34      0.782      0.725      0.803      0.489

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      20/30      5.22G      1.255      1.529      1.183         33        640: 100% ━━━━━━━━━━━━ 14/14 1.5it/s 9.6s0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.7it/s 0.4s
                   all         20         34      0.774      0.628      0.737       0.45
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      21/30      5.22G      1.135      1.726      1.169         17        640: 100% ━━━━━━━━━━━━ 14/14 1.3it/s 11.2s0.8s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.5it/s 0.4s
                   all         20         34      0.808      0.742      0.808      0.475

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      22/30      4.32G      1.122      1.614      1.187         23        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      22/30      5.26G      1.157       1.67      1.169         17        640: 100% ━━━━━━━━━━━━ 14/14 1.3it/s 10.9s0.8s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.3it/s 0.4s
                   all         20         34      0.692      0.723      0.744       0.45

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      23/30      5.25G      1.082      1.557      1.133         13        640: 100% ━━━━━━━━━━━━ 14/14 1.4it/s 9.9s0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.7it/s 0.4s
                   all         20         34      0.562      0.714      0.675      0.444

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      24/30      5.25G      1.103      1.566      1.135         16        640: 100% ━━━━━━━━━━━━ 14/14 1.5it/s 9.5s0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.5it/s 0.4s
                   all         20         34      0.733      0.651      0.785      0.504

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      25/30      4.26G      1.143      1.596      1.173         23        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      25/30      5.25G       1.07      1.467      1.123         15        640: 100% ━━━━━━━━━━━━ 14/14 1.4it/s 10.3s0.8s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.7it/s 0.4s
                   all         20         34      0.753      0.764      0.838      0.559

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      26/30      4.25G      0.973      1.618      1.187         21        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      26/30      5.26G      1.068      1.473      1.112         17        640: 100% ━━━━━━━━━━━━ 14/14 1.5it/s 9.6s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.8it/s 0.4s
                   all         20         34      0.868      0.717      0.864      0.588

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      27/30      5.26G      1.024      1.421      1.108         14        640: 100% ━━━━━━━━━━━━ 14/14 1.5it/s 9.3s0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.8it/s 0.4s
                   all         20         34      0.861      0.833      0.868       0.57

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      28/30      4.27G     0.8577      2.001      1.185         17        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      28/30      4.28G      1.007      1.425      1.093         16        640: 100% ━━━━━━━━━━━━ 14/14 1.4it/s 10.2s0.8s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.4it/s 0.4s
                   all         20         34      0.897      0.815      0.847       0.55

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      29/30      4.26G      1.006      1.396      1.071         15        640: 100% ━━━━━━━━━━━━ 14/14 1.4it/s 10.1s0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.8it/s 0.4s
                   all         20         34      0.892      0.811      0.847      0.561

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      30/30      4.25G      1.001      1.385      1.113         22        640: 0% ──────────── 0/14  0.7s

scatter_reduce_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)
index_put_with_accumulate_mps does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at /Users/runner/work/pytorch/pytorch/aten/src/ATen/Context.cpp:190.)


      30/30      5.23G     0.9744      1.415      1.075         13        640: 100% ━━━━━━━━━━━━ 14/14 1.5it/s 9.4s0.7s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 2.7it/s 0.4s
                   all         20         34      0.892      0.813      0.847      0.549

30 epochs completed in 0.093 hours.
Optimizer stripped from /Users/emmanuel/DeVeLoP/AI-ML/doom-mlg-ai/runs/detect/train-2/weights/last.pt, 5.5MB
Optimizer stripped from /Users/emmanuel/DeVeLoP/AI-ML/doom-mlg-ai/runs/detect/train-2/weights/best.pt, 5.5MB

Validating /Users/emmanuel/DeVeLoP/AI-ML/doom-mlg-ai/runs/detect/train-2/weights/best.pt...
Ultralytics 8.4.155 🚀 Python-3.14.3 torch-2.14.0 MPS (Apple M4)
YOLO11n summary (fused): 100 layers, 2,583,322 parameters, 0 gradients, 6.4 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 1.6it/s 0.6s
                   all         20         34  

## Results:

The model over all didn't reached a crazy 10/10 classification but reached a 0.864 mAP50 which means that the prediction box over the enemy had at least average precision of .50.


### It's going to be always precise?
No, definitely no, but it's a great aproach for only 290 labeled images, wehere some classes were underrepresented over others.